common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -374,17 +352,7 @@ private void record0(Object hint, int recordsToSkip) {
 
         @Override
         public boolean close() {
-            if (freed.compareAndSet(false, true)) {
-                synchronized (head) {
-                    active --;
-                    prev.next = next;
-                    next.prev = prev;
-                    prev = null;
-                    next = null;
-                }
-                return true;
-            }
-            return false;
+            return allLeaks.remove(this, Boolean.TRUE);
```

Comment:
I could, but that will call AbstractMap.remove which creates an iterator, and then calls the underlying remove.   This skips that step.